### ISSUE TITLE
Fix request: False positive pmd:AvoidUnusedAssignments

### ISSUE THREAD

**Issue by koosg** [jborgers/PMD-jPinpoint-rules#241]
The following also gives a false positive. Possibly because of using `+=`
```
public static boolean elfProef(final String accountNumber) {
    int sum = 0;
    final int length = accountNumber.length();
    for (int i = 0; i < length; ++i) {
        sum += Character.digit(accountNumber.charAt(i), RADIX) * (length - i);
    }
    return sum % DIVIDER == 0;
}
```
It complains that the assignment to int sum=0 should be avoided...
